kommunicate/src/main/java/io/kommunicate/zendesk/KmZendeskClientService.java

```diff
@@ -27,6 +28,55 @@ public KmZendeskClientService(Context context) {
     private String getKmZendeskJwtUrl() {
         return getKmBaseUrl() + KM_ZENDESK_JWT_URL;
     }
+    private String getKmZendeskSendMessageUrl() { return getKmBaseUrl() + KM_ZENDESK_SEND_MESSAGE; }
+    private String getKmZendeskSendAttachmentUrl() { return getKmBaseUrl() + KM_ZENDESK_SEND_ATTACHMENT; }
+
+    public KmZendeskApiModel sendZendeskMessage(String message, String displayName, String agentId, Integer conversationId, Long messageTimestamp) {
+        try {
+            JSONObject messageProxy = new JSONObject();
+            JSONObject agentInfo = new JSONObject();
+            agentInfo.put("displayName", displayName);
+            agentInfo.put("agentId", agentId);
+                    messageProxy.put("message", message);
+                    messageProxy.put("groupId", conversationId);
+                    messageProxy.put("fromUserName", agentId);
+                    messageProxy.put("messageDeduplicationKey", agentId + "-" + messageTimestamp);
+                    messageProxy.put("agentInfo", agentInfo);
+
+
+            String response = httpRequestUtils.postData(getKmZendeskSendMessageUrl(), "application/json", "application/json", messageProxy.toString());
+            KmZendeskApiModel jwtResponse = (KmZendeskApiModel) GsonUtils.getObjectFromJson(response,
+                    KmZendeskApiModel.class);
+            Utils.printLog(context, TAG, "Zendesk Send Agent Message response : " + response);
+            return jwtResponse;
+        } catch (Exception e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+    public KmZendeskApiModel sendZendeskAttachment(Attachment attachment, String displayName, String agentId, Integer conversationId, Long messageTimestamp) {
+        try {
+            JSONObject messageProxy = new JSONObject();
```

Comment:
most of the code for agentInfo and messageProxy object creation looks same as present in above method, move common things to reuse code.

Reply:
created a createMessageProxy() method to handle common object creation